index.js

```diff
@@ -175,6 +172,78 @@ const parseArgs = (
   return result;
 };
 
+/**
+ * Determines if `arg` is a stdin/stdout positional.
+ * See [Guideline 13](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html) of the 12.2 Utility Syntax Guidelines.
+ * @example '-'
+ */
+function isStdInOutPositional(arg) {
+  return arg === '-';
+}
+
+/**
+ * Determines if `arg` is an options terminator.
+ * See [Guideline 10](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html) of the 12.2 Utility Syntax Guidelines.
+ * @example '--'
+ */
+function isOptionsTerminator(arg) {
+  return arg === '--';
+}
+
+/**
+ * Determines if `arg` is an option.
+ * See [Guideline 4](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html) of the 12.2 Utility Syntax Guidelines.
+ * @example '-f' | '-foo' | '--foo'
+ */
+function isOption(arg) {
+  return StringPrototypeStartsWith(arg, '-');
+}
+
+/**
+ * Determines if `arg` is a short option.
+ * See [Guideline 5](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html) of the 12.2 Utility Syntax Guidelines.
+ * @example '-f' | '-foo'
+ */
+function isShortOption(arg) {
+  return isOption(arg) &&
+    StringPrototypeCharAt(arg, 1) !== '-';
+}
+
+/**
+ * Determines if `arg` is a short group option.
+ * See [Guideline 5](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html) of the 12.2 Utility Syntax Guidelines.
+ * @example '-foo'
+ */
+function isShortGroupOption(arg) {
+  return isShortOption(arg) && arg.length > 2;
+}
+
+/**
+ * Determines if the option explicitly sets an argument.
+ * Identified by the presence of `=` in the arg.
+ * @example '--foo=bar'
+ */
+function hasExplicitOptionArgument(arg) {
+  return StringPrototypeIncludes(arg, '=');
+}
+
+/**
+ * Determines if the `arg` following a given index is an option.
+ * @example
+ * isNextArgOption(['--foo', '--bar'], 0) // true
+ */
+function isNextArgOption(argv, pos) {
+  return pos + 1 < argv.length && isOption(argv[pos + 1]);
+}
+
+parseArgs.isStdInOutPositional = isStdInOutPositional;
+parseArgs.isOptionsTerminator = isOptionsTerminator;
+parseArgs.isOption = isOption;
```

Comment:
This expands the API surface quite a bit, which I think is a hindrance to the goal of a tight MVP, that we can get merged into the Node.js codebase with minimal controversy.

I'd rather we add these as private helpers if they add value, but don't expose on the `parseArgs` function.